### ISSUE TITLE
Visibility of modules regarding notations, implicit morphisms

### ISSUE THREAD

**Issue by florian-rabe** [UniFormal/MMT#345]
I agree on notations for views.
In #343 @kohlhase asks about notations for views, for which I am creating a separare issue here:

BTW, one thing I would also like is "notations for views", i.e. I would
like to write something like

```
view SomeThingLongAndUglyButSystematic : ?A -> ?B # short= ... ||
```
and then use the syntax "short A" for the application of this view to A.

**Comment by florian-rabe** [comment]
Similarly, parametric theories should have them.
Some parts of that are implemented already.

But I got stuck with the collection of notations: parsing needs to know in advance which notations may come up, but views and theories are global. So it's unclear which notations to collect.

A related problem is posed by implicit morphisms, where MMT also needs to know in advance which ones may come up.

I think some experimental feedback on this question is needed.
Example solutions:

* consider all theories/morphisms from the same folder/archive
* consider all theories/morphisms from the same namespace (but possibly in different archives?)
* new syntax that lets users spell out which theories/morphisms to use (exclusively or additionally?)

**Comment by kohlhase** [comment]
Ah, yes, global objects and notations. 

> I think some experimental feedback on this question is needed.

I am not sure how you would do that. How about implementing 3. and at some point going over usage patterns to see whether 1. or 2. can replace most of 3. syntax.

**Comment by kohlhase** [comment]
> Ah, yes, global objects and notations.

could you not just collect all notations from views (transitively) into the current theory and all imported parametric theories (after all we only want to use the notations in "applied form"). Maybe only for implicit morphisms?

**Comment by florian-rabe** [comment]
Ah, yes. For views that can be an option (not so much for parametric theories).

That still would leave the questions from which set to collect views:

1 everything currently loaded into memory
1 everything in a currently opened archive
1 all known archives

Probably 2 is best.
But that'll require some better management of the relational info in an archive:

* must included view notations
* must be loaded more efficiently

**Comment by kohlhase** [comment]
I agree that 2. is probably best.

**Comment by kohlhase** [comment]
> In any case, I'm worried the set of available notations becomes too big and an entirely different mechanism will be needed to handle ambiguities.

I do not think that adding view notations and parameterized theory notations will make such a big difference; we do not have sooo many (yet)

> Especially, considering, I also want every notation from an implicitly visible theory to be visible.

that might indeed make a difference.
